prov/util/src/util_mem_hooks.c

```diff
@@ -553,56 +600,27 @@ static int ofi_memhooks_start(struct ofi_mem_monitor *monitor)
 	for (i = 0; i < OFI_INTERCEPT_MAX; ++i)
 		dlist_init(&intercepts[i].dl_intercept_list);
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MMAP]);
-	if (ret) {
-		FI_WARN(&core_prov, FI_LOG_MR,
-		       "intercept mmap failed %d %s\n", ret, fi_strerror(ret));
-		return ret;
-	}
-
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MUNMAP]);
-	if (ret) {
-		FI_WARN(&core_prov, FI_LOG_MR,
-		       "intercept munmap failed %d %s\n", ret, fi_strerror(ret));
-		return ret;
-	}
-
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MREMAP]);
-	if (ret) {
-		FI_WARN(&core_prov, FI_LOG_MR,
-		       "intercept mremap failed %d %s\n", ret, fi_strerror(ret));
-		return ret;
-	}
-
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MADVISE]);
-	if (ret) {
-		FI_WARN(&core_prov, FI_LOG_MR,
-		       "intercept madvise failed %d %s\n", ret, fi_strerror(ret));
-		return ret;
-	}
-
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_SHMAT]);
-	if (ret) {
-		FI_WARN(&core_prov, FI_LOG_MR,
-		       "intercept shmat failed %d %s\n", ret, fi_strerror(ret));
-		return ret;
-	}
-
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_SHMDT]);
-	if (ret) {
-		FI_WARN(&core_prov, FI_LOG_MR,
-		       "intercept shmdt failed %d %s\n", ret, fi_strerror(ret));
-		return ret;
+	i_err = -1;
+	for (i = 0; i < OFI_INTERCEPT_MAX; ++i) {
+		ret = ofi_intercept_symbol(&intercepts[i]);
+		if (ret != 0) {
+			i_err = i;
+			FI_DBG(&core_prov, FI_LOG_MR,
+				"intercept %s failed %d %s\n", intercepts[i].symbol,
+					ret, fi_strerror(ret));
+			break;
+		}
 	}
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_BRK]);
-	if (ret) {
-		FI_WARN(&core_prov, FI_LOG_MR,
-		       "intercept brk failed %d %s\n", ret, fi_strerror(ret));
-		return ret;
+	if (i_err != -1) {
+		for (i = 0; i <= i_err; ++i) {
```

Comment:
shouldn't it be "i < i_err"? Since at time i_err was set, ofi_intercept_symbol failed in that iteration.

Reply:
I think the cleaner approach would be to jump to an err label, then do something like:
```
while (--i >= 0)
    unwind...
```
Otherwise, the code ends up checking for an error twice.

Reply:
Thank you Sean - that does look better and I have reflected that change in latest.

Reply:
> shouldn't it be "i < i_err"? Since at time i_err was set, ofi_intercept_symbol failed in that iteration.

You are correct, thank you. I also changed approaches (to Sean's approach) so this is no longer part of the PR.